### PR TITLE
Fixed invocations of startGoRoutine

### LINKED ISSUE #960: Misleading references to gateway.go in stack
When getting a stack, the routines for readLoop and writeLoop reference gateway.go, even when no gateway is running.
For instance, starting `gnatsd` and having one client connected, then getting the stack shows something like this:
```
goroutine 24 [sync.Cond.Wait]:
sync.runtime_notifyListWait(0xc00013d210, 0x0)
	/usr/local/go/src/runtime/sema.go:510 +0xeb
sync.(*Cond).Wait(0xc00013d200)
	/usr/local/go/src/sync/cond.go:56 +0x92
github.com/nats-io/gnatsd/server.(*client).writeLoop(0xc0001ac000)
	/Users/ivan/dev/go/src/github.com/nats-io/gnatsd/server/client.go:642 +0x12d
github.com/nats-io/gnatsd/server.(*client).writeLoop-fm()
	/Users/ivan/dev/go/src/github.com/nats-io/gnatsd/server/gateway.go:703 +0x2a
created by github.com/nats-io/gnatsd/server.(*Server).startGoRoutine
	/Users/ivan/dev/go/src/github.com/nats-io/gnatsd/server/server.go:1872 +0x94
```
There should not be reference to `gateway.go` in that context.

### PR DESCRIPTION
**Pull request by kozlovic**
Resolves #960

### CODE REVIEW

File: server/server.go [hunk 1]
```diff
@@ -1866,6 +1872,12 @@ func (s *Server) ID() string {
 }
 
 func (s *Server) startGoRoutine(f func()) {
+	if checkStartGoRoutineFunc {
+		name := runtime.FuncForPC(reflect.ValueOf(f).Pointer()).Name()
+		if strings.HasSuffix(name, "-fm") {
```
Comment (derekcollison) [inline server/server.go]: This stable across releases?

Comment (kozlovic) [reply server/server.go]: You mean go releases, right? not sure. the alternative is I could check for end of name being `.funcxxx`. It is sometimes `.func2()` or similar. I could dig a bit more.
Works for go 1.11 and 1.12. We could adapt in the future too.

Comment (derekcollison) [reply server/server.go]: ok, just not sure how important it is to have the check etc.

Comment (kozlovic) [reply server/server.go]: I can definitively remove check (and therefore test). Let me know if you want me to do that.

Comment (derekcollison) [reply server/server.go]: For now I would.

Comment (kozlovic) [reply server/server.go]: Ok removed check and test. Did a push --force.

**Pull request merged**